File: src/main/java/pl/mjedynak/idea/plugins/builder/factory/PopupChooserBuilderFactory.java
package pl.mjedynak.idea.plugins.builder.factory;

import com.intellij.openapi.ui.popup.PopupChooserBuilder;
import javax.swing.JList;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

public class PopupChooserBuilderFactory {

    private PopupChooserBuilderFactory() {
        throw new UnsupportedOperationException("Utility class");
    }

    @Contract("_ -> new")
    public static @NotNull PopupChooserBuilder<?> getPopupChooserBuilder(@NotNull JList<?> list) {
        return new PopupChooserBuilder<>(list);
    }
}

File: src/main/kotlin/io/github/kingg22/kobra/builder/factory/PopupChooserBuilderFactory.kt
package io.github.kingg22.kobra.builder.factory

import com.intellij.openapi.ui.popup.PopupChooserBuilder
import org.jetbrains.annotations.Contract
import javax.swing.JList

public object PopupChooserBuilderFactory {
    @Contract("_ -> new")
    @JvmStatic
    public fun getPopupChooserBuilder(list: JList<*>): PopupChooserBuilder<*> = PopupChooserBuilder(list)
}
